### PR TITLE
Extend catalog test script to run tests for different platforms

### PR DESCRIPTION
**Pull request by barthy1**
# Changes

PLATFORM env variable is introduced. This variable allows to run tests for specific platform (for instance, "linux/s390x" or "linux/ppc64le").
If the environment variable is specified:
- some tests are updated with different image values, required for specific platform.
Image names should be specified in script file, separate for each platform. In the current code script file for linux/s390x is proposed.
- only tasks, which have mention about corresponding platform support in the annotation, will be tested.
If PLATFORM env variable is not specified, the behaviour of test scripts will be similar to the one before the PR.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: test/e2e-common.sh [hunk 1]
```diff
@@ -33,6 +33,15 @@ if [ "${BASH_VERSINFO:-0}" -lt 4 ];then
     exit 1
 fi
 
+# Default registry image value to use in this script
+REGISTRY_IMAGE="registry"
+
+# Do the tasks modifications if special PLATFORM value is specified. By default nothing happens.
+if [[ -n ${PLATFORM} ]] && [[ -f "$(dirname $0)/$(echo ${PLATFORM}| tr / -).sh" ]]; then
+        # Load script specific to platform. File name should follow the pattern "os-arch.sh", for instance "linux-s390x.sh".
```
Comment (chmouel) [inline test/e2e-common.sh:41]: 👍🏻

File: test/e2e-common.sh [hunk 2]
```diff
@@ -166,6 +175,10 @@ function test_yaml_can_install() {
         for ignore in ${TEST_YAML_IGNORES};do
             [[ ${ignore} == "${testname}" ]] && skipit=True
         done
+
+        # In case if PLATFORM env variable is specified, do the tests only for matching tasks
+        [[ -n ${PLATFORM} ]] && [[ $(grep "tekton.dev/platforms" ${runtest}) != *"${PLATFORM}"* ]]  && skipit=True
```
Comment (chmouel) [inline test/e2e-common.sh:180]: maybe redirect stderr to /dev/null since it may fails (but the case is handled in skipit)

**Review by chmouel — COMMENTED** [review]
so far so good thanks, i have added some minor comments

File: test/e2e-common.sh (hunk 2 above)
Comment (barthy1) [reply test/e2e-common.sh]: @chmouel done

**Comment by chmouel** [comment]
/lgtm
thank you!

**Review by vinamra28 — APPROVED** [review]

**Pull request merged**